File: python/6_kyu/dubstep.py
# Dubstep
# https://www.codewars.com/kata/551dc350bf4e526099000ae5/train/python

# Solution 1: Split, Remove, Join
def song_decoder(s):
    s = s.split("WUB")
    while "" in s:
        s.remove("")
    return " ".join(s)

# Solution 2: Pythonic Split, Replace, Join
def song_decoder(s):
    return " ".join(s.replace("WUB", " ").split())

print(song_decoder("AWUBBWUBC"))
print(song_decoder("AWUBWUBWUBBWUBWUBWUBC"))
print(song_decoder("WUBAWUBBWUBCWUB"))
print(song_decoder("WUBWEWUBAREWUBWUBTHEWUBCHAMPIONSWUBMYWUBFRIENDWUB"))

File: javascript/6_kyu/dubstep.js
// Dubstep
// https://www.codewars.com/kata/551dc350bf4e526099000ae5/train/javascript

// Solution 1: Split, Filter, Join
function songDecoder(s) {
    s = s.split("WUB");

    filteredSong = s.filter(function (x) {return x != ""});

    return filteredSong.join(" ");
}

// Solution 2: Split, Filter boolean, join
function songDecoder(s) {
    return s.split("WUB").filter(Boolean).join(" ");
}

console.log(songDecoder("AWUBBWUBC"));
console.log(songDecoder("AWUBWUBWUBBWUBWUBWUBC"));
console.log(songDecoder("WUBAWUBBWUBCWUB"));
console.log(songDecoder("WUBWEWUBAREWUBWUBTHEWUBCHAMPIONSWUBMYWUBFRIENDWUB"));